clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -393,17 +340,20 @@ private void throwIfRecordTooLarge() throws RecordTooLargeException {
             return Collections.emptyMap();
         } else {
             Map<TopicPartition, List<ConsumerRecord<K, V>>> drained = new HashMap<>();
-            throwIfOffsetOutOfRange();
-            throwIfUnauthorizedTopics();
-            throwIfRecordTooLarge();
-
             int maxRecords = maxPollRecords;
```

Comment:
Yeah, that could work. It could be confused with a collection having records to drain, but it's a local variable, so I'll just let you pick something (you can keep the current name if you like as well). :)